app/customer.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+import math
+import datetime
+from dataclasses import dataclass
+from app.car import Car
+from app.shop import Shop
+
+
+class NotEnoughMoney(Exception):
+    pass
+
+
+@dataclass()
+class Customer:
+    name: str
+    product_cart: dict
+    location: list
+    money: int
+    car: Car
+
+    def get_info_customer(self) -> None:
+        print(f"{self.name} has {self.money} dollars")
+
+    def calculate_cheapest_trip(self, shops: list, fuel_price: float) -> tuple:
+        travel_costs = {}
+        for shop in shops:
+            cost_road = self.calculate_cost_road(shop, fuel_price)
+            cost_products = self.calculate_cost_products(shop.products)
+            costs = cost_road + cost_products
+            print(f"{self.name}'s trip to the {shop.name} costs {costs}")
+            travel_costs[shop] = costs
+        cheapest_shop, cheapest_price = (
+            min(travel_costs.items(), key=lambda x: x[1])
+        )
+        if self.money < cheapest_price:
+            print(f"{self.name} doesn't have enough money "
+                  f"to make a purchase in any shop")
+            raise NotEnoughMoney
+        print(f"{self.name} rides to {cheapest_shop.name}")
+        print("")
+        return cheapest_shop, cheapest_price
+
+    def calculate_cost_products(self, price: float) -> float:
+        cost_products = 0
+        for product in self.product_cart:
+            cost_products += price[product] * self.product_cart[product]
+        return cost_products
```

Comment:
You can calculate the total cost of products in one line using a list comprehension